### PR TITLE
backend/feat: #1877 post ride safety

### PR DESCRIPTION
**Pull request by PraveenGongada**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Added `is_safe` and `issue_id` fields to driver rating table for night ride safety

### Additional Changes

- [x] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Rating.hs [hunk 1]
```diff
@@ -31,13 +31,15 @@ import qualified Storage.Beam.Rating as BeamR
 create :: (L.MonadFlow m, Log m) => DR.Rating -> m ()
 create = createWithKV
 
-updateRating :: (L.MonadFlow m, MonadTime m, Log m) => Id Rating -> Id Person -> Int -> Maybe Text -> m ()
-updateRating (Id ratingId) (Id driverId) newRatingValue newFeedbackDetails = do
+updateRating :: (L.MonadFlow m, MonadTime m, Log m) => Id Rating -> Id Person -> Int -> Maybe Text -> Maybe Text -> Bool -> m ()
+updateRating (Id ratingId) (Id driverId) newRatingValue newFeedbackDetails newIssueId newStatus = do
```
[the remark this answers is not in the document]
Comment (PraveenGongada) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Rating.hs:35]: Here it is being used as text, why ID type when we are not using it as ID?

**Comment by 0utkarsh** [comment]
- After completing a night ride (between 10 PM and 6 AM), customers will see a safety popup on their app screen.
- If the customer selects "Unsafe," the support center will be notified immediately.

The logics for above will be handled from frontend? What about the following?

- The customer will be given the option to select specific issues they encountered during the ride (e.g., inappropriate behavior, route deviation, reckless driving, etc.).

**Review by 0utkarsh — CHANGES_REQUESTED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/Storage/Beam/Issue.hs [hunk 2]
```diff
@@ -38,6 +38,7 @@ data IssueT f = IssueT
     description :: B.C f Text,
     ticketId :: B.C f (Maybe Text),
     status :: B.C f Domain.IssueStatus,
+    rideId :: B.C f (Maybe Text),
```
[the remark this answers is not in the document]
Comment (PraveenGongada) [reply Backend/app/rider-platform/rider-app/Main/src/Storage/Beam/Issue.hs:41]: We can't use Id types here after KV changes right?

[the remark this answers is not in the document]
Comment (0utkarsh) [reply Backend/app/rider-platform/rider-app/Main/src/Storage/Beam/Issue.hs:41]: Sure, I am approving this PR, can you answer the above comment too? We can merge it then

**Review by 0utkarsh — DISMISSED** [review]

**Comment by PraveenGongada** [comment]
> 

yes frontend will show pop up and send the specific issue in the sendIssue api

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Support.hs [hunk 3]
```diff
@@ -60,15 +61,19 @@ validateIssue Issue {..} =
...
   }
   deriving (Generic, Show, FromJSON, ToJSON, ToSchema)
 
 validateSendIssueReq :: Validate SendIssueReq
 validateSendIssueReq SendIssueReq {..} =
   validateObject "issue" issue validateIssue
 
-type SendIssueRes = APISuccess
+newtype SendIssueRes = SendIssueRes
+  { issueId :: Text
```
Comment (hkmangla) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Support.hs:74]: Not a backward compatible change.

**Review by hkmangla — CHANGES_REQUESTED** [review]

**Review by soumyajit2304 — APPROVED** [review]

**Pull request closed without merging**